src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+type ShapeType = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ShapeType;
+  color: string;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: ShapeType = 'triangle';
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const s = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.round(s * 100) / 100;
+  }
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier (like public or private), TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to manually assign the values like `this.a = a;`. Since you are using `public` in the constructor parameters, the manual assignments can be removed.